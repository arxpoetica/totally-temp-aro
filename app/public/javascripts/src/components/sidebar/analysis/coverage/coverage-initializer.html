<style scoped>
  #table-coverage-initializer td {
    line-height: 33px;  /* To vertically align text */
  }
</style>
<!-- <p>
  I am an AngularJS component. Value from Redux: {{$ctrl.value}}
  <button class="btn btn-light" ng-click="$ctrl.increment()">Update from AngularJS</button>
</p> -->
<table id="table-coverage-initializer"  ng-if="$ctrl.isLoggedInUserSuperUser" class="table table-sm table-striped">
  <tbody>

    <!-- Coverage type -->
    <tr>
      <td>Coverage Type</td>
      <td>
        <select class="form-control"
                ng-model="$ctrl.state.coverage.initializationParams.coverageType"
                ng-options="item.id as item.name for item in $ctrl.coverageTypes">
        </select>
      </td>
    </tr>

    <!-- Save site coverage -->
    <tr ng-if="$ctrl.state.coverage.initializationParams.coverageType === 'location'">
      <td>Save site coverage</td>
      <td>
        <input type="checkbox"
               class="checkboxfill"
               ng-model="$ctrl.state.coverage.initializationParams.saveSiteCoverage">
      </td>
    </tr>
    
    <!-- Limit To Marketable Technology -->
    <tr>
      <td>Limit To Marketable Technology</td>
      <td>
        <input type="checkbox"
               class="checkboxfill"
               ng-model="$ctrl.state.coverage.initializationParams.useMarketableTechnologies">
      </td>
    </tr>

    <!-- Limit To Max Equipment Speed -->
    <tr>
      <td>Limit To Max Equipment Speed</td>
      <td>
        <input type="checkbox"
               class="checkboxfill"
               ng-model="$ctrl.state.coverage.initializationParams.useMaxSpeed">
      </td>
    </tr>

    <!-- Site assignment -->
    <tr>
      <td>Site Assignment</td>
      <td>
        <select class="form-control"
                ng-model="$ctrl.selectedSiteAssignment"
                ng-options="item as item for item in $ctrl.siteAssignments">
        </select>
      </td>
    </tr>

    <!-- The selection type (service areas/analysis areas) -->
    <tr>
        <td>Selection Type</td>
        <td>
          <select class="form-control" 
                  ng-model="$ctrl.state.optimizationOptions.analysisSelectionMode"
                  ng-change="$ctrl.onSelectionTypeChange($ctrl.state.optimizationOptions.analysisSelectionMode)">
            <option ng-repeat="(selectionId, selectionVal) in $ctrl.allowedSelectionModes" value="{{selectionVal}}">{{$ctrl.selectionModeLabels[selectionVal]}}</option>
          </select>
        </td>
      </tr>
    <tr ng-if="$ctrl.state.optimizationOptions.analysisSelectionMode === $ctrl.state.selectionModes.SELECTED_AREAS">
      <td>Selected Geographies({{$ctrl.serviceAreas.length || 0}})
        <a ng-if="$ctrl.serviceAreas.length > 0" ng-click="$ctrl.removeServiceAreas($ctrl.serviceAreas)">
          <span class="far fa-trash-alt trash"></span>
        </a>
      </td>
      <td> 
        <p ng-hide="$ctrl.serviceAreas.length > 0"><em>No geographies selected</em></p>
        <show-targets targets="$ctrl.serviceAreas"
                      remove-target="$ctrl.removeServiceAreas(target)"></show-targets>
      </td>
    </tr>
    <tr ng-if="$ctrl.state.optimizationOptions.analysisSelectionMode === $ctrl.state.selectionModes.SELECTED_ANALYSIS_AREAS">
      <td>Selected Analysis Areas({{$ctrl.analysisAreas.length || 0}})
        <a ng-if="$ctrl.analysisAreas.length > 0" ng-click="$ctrl.removeAnalysisAreas($ctrl.analysisAreas)">
          <span class="far fa-trash-alt trash"></span>
        </a>
      </td>
      <td>
        <p ng-hide="$ctrl.analysisAreas.length > 0"><em>No analysis areas selected</em></p>
        <show-targets targets="$ctrl.analysisAreas"
                      remove-target="$ctrl.removeAnalysisAreas(target)" zoom-target="$ctrl.zoomTarget({target:$ctrl.targets})"></show-targets>
      </td>
    </tr>
  </tbody>
</table>

<!-- If we have a valid report, then disable all controls -->
<div ng-if="$ctrl.isLoggedInUserSuperUser && $ctrl.state.coverage.report"
     class="disable-sibling-controls">

</div>
        
<div class="alert alert-danger" ng-if="!$ctrl.isLoggedInUserSuperUser" style="margin-top: 60px;">
  You must be a system-wide superuser in order to use coverage reports.
</div>
<!-- Spinner that will show up if the UI configuration is being loaded from the server -->
<!-- <div ng-show="$ctrl.state.coverage.isInitializingReport" style="position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; text-align: center; background-color: rgba(255, 255, 255, 0.7); z-index: 3;">
  <i class="fa fa-spinner fa-spin" style="font-size:48px; position: relative; top: 50%; transform: translateY(-50%); "></i>
</div> -->