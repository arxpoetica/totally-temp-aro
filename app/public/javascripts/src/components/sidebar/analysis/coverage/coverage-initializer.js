import { createSelector } from 'reselect'

const getAllSelectionModes = state => state.selection.selectionModes
const getAllowedSelectionModes = createSelector([getAllSelectionModes], 
  (selectionModes) => angular.copy(selectionModes.filter(item => item.id !== 'SELECTED_LOCATIONS')))

class CoverageInitializerController {
  constructor(state, $http, $timeout, $ngRedux) {
    this.state = state
    this.$http = $http
    this.$timeout = $timeout
    this.unsubscribeRedux = $ngRedux.connect(this.mapStateToThis, this.mapDispatchToTarget)(this)
    this.coverageTypes = [
      { id: 'census_block', name: 'Form 477' },
      { id: 'location', name: 'Locations' }
    ]

    this.serviceAreas = []
    this.analysisAreas = []
    this.siteAssignments = ['Proximity', 'Incremental']
    this.selectedSiteAssignment = 'Proximity'
  }

  // Map global state to component properties
  mapStateToThis(state) {
    return {
      isSuperUser: state.user.isSuperUser,
      activeSelectionModeId: state.selection.activeSelectionMode.id,
      selectionModes: getAllowedSelectionModes(state),
      coverageType: state.coverage.initializationParams.coverageType,
      saveSiteCoverage: state.coverage.initializationParams.saveSiteCoverage,
      useMarketableTechnologies: state.coverage.initializationParams.useMarketableTechnologies,
      useMaxSpeed: state.coverage.initializationParams.useMaxSpeed,
      coverageReport: state.coverage.report
    }
  }

  mapDispatchToTarget(dispatch) {
    return { }
  }

  $onDestroy() {
    this.unsubscribeRedux()
  }

  onSelectionTypeChange(selectionType) {
    this.state.selectionTypeChanged.next(selectionType)
  } 

  removeServiceAreas(targets) {
    this.$http.post(`/service_areas/${this.state.plan.getValue().id}/removeServiceAreaTargets`, { serviceAreaIds: targets.map((sa) => sa.id) })
      .then((response) => {
        this.state.reloadSelectedServiceAreas()
      })
      .catch(err => console.error(err))
  }

  removeAnalysisAreas(targets) {
    this.$http.post(`/analysis_areas/${this.state.plan.getValue().id}/removeAnalysisAreaTargets`, { analysisAreaIds: targets.map((sa) => sa.id) })
      .then((response) => {
        this.state.reloadSelectedAnalysisAreas()
      })
      .catch(err => console.error(err))
  }

  $onChanges(changesObj) {
    if (changesObj.selection) {
      // The selected service areas have changed.
      var serviceAreaIds = Array.from(this.state.selection.planTargets.serviceAreaIds)
      this.$http.post('/network_plan/service_area/addresses', { serviceAreaIds: serviceAreaIds })
        .then((result) => {
          this.serviceAreas = result.data
        })
        .catch(err => console.error(err))
      
      // The selected analysis areas have changed.
      var analysisAreaIds = Array.from(this.state.selection.planTargets.analysisAreaIds)
      this.$http.post('/network_plan/analysis_area/addresses', { analysisAreaIds: analysisAreaIds })
        .then((result) => {
          this.analysisAreas = result.data
        })
        .catch(err => console.error(err))
    }
  }
}

CoverageInitializerController.$inject = ['state', '$http', '$timeout', '$ngRedux']

let coverageInitializer = {
  templateUrl: '/components/sidebar/analysis/coverage/coverage-initializer.html',
  bindings: {
    selection: '<'
  },
  controller: CoverageInitializerController
}

export default coverageInitializer
