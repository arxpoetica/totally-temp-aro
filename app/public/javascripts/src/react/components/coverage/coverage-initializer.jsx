import React from 'react'
import { Component } from 'react'
import { PropTypes } from 'prop-types'

export default class CoverageInitializer extends Component {
  render() {
    return <div>
      <p>Hello world</p>
      <p>Property1: {this.props.property1}</p>
    </div>
  }
}

CoverageInitializer.propTypes = {
  property1: PropTypes.string
}
