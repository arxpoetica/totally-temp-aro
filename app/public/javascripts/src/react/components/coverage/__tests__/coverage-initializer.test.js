/* global test expect jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { CoverageInitializer } from '../coverage-initializer'
import SelectionModes from '../../selection/selection-modes'

// Constants to use throughout the tests
const selectionModes = [
  { id: SelectionModes.SELECTED_AREAS, description: 'Service Areas' },
  { id: SelectionModes.SELECTED_LOCATIONS, description: 'Locations' },
  { id: SelectionModes.SELECTED_ANALYSIS_AREAS, description: 'Analysis Areas' }
]

// -----------------------------------------------------------------------------
test('When the logged in user is not a SuperUser', () => {
  const component = renderer.create(
    <CoverageInitializer isSuperUser={false}
      selectionModes={selectionModes}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When the logged in user is a SuperUser', () => {
  const component = renderer.create(
    <CoverageInitializer isSuperUser
      selectionModes={selectionModes}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When we do not have an existing report', () => {
  const component = renderer.create(
    <CoverageInitializer isSuperUser
      selectionModes={selectionModes}
      coverageReport={null}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When we have an existing report', () => {
  const component = renderer.create(
    <CoverageInitializer isSuperUser
      selectionModes={selectionModes}
      coverageReport={{}}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When we have supplied all parameters', () => {
  const component = renderer.create(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When the coverage type is changed', () => {
  const mockSetCoverageType = jest.fn()
  const component = shallow(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
      setCoverageType={mockSetCoverageType}
    />
  )
  component.find('#selectCoverageType').simulate('change', { target: { value: 'census_block' } })
  expect(mockSetCoverageType.mock.calls[0][0]).toEqual('census_block')
})

// -----------------------------------------------------------------------------
test('When the group key type is changed', () => {
  const mockSetGroupKeyType = jest.fn()
  const component = shallow(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
      setGroupKeyType={mockSetGroupKeyType}
    />
  )
  component.find('#selectGroupKeyType').simulate('change', { target: { value: 'location' } })
  expect(mockSetGroupKeyType.mock.calls[0][0]).toEqual('location')
})

// -----------------------------------------------------------------------------
test('When the limit-marketable-technology flag is changed', () => {
  const mockSetLimitMarketableTechnology = jest.fn()
  const component = shallow(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
      setLimitMarketableTechnology={mockSetLimitMarketableTechnology}
    />
  )
  component.find('#chkLimitMarketableTechnologies').simulate('change', { target: { checked: false } })
  expect(mockSetLimitMarketableTechnology.mock.calls[0][0]).toEqual(false)
})

// -----------------------------------------------------------------------------
test('When the limit-max-speed flag is changed', () => {
  const mockSetLimitMaxSpeed = jest.fn()
  const component = shallow(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
      setLimitMaxSpeed={mockSetLimitMaxSpeed}
    />
  )
  component.find('#chkLimitMaxSpeed').simulate('change', { target: { checked: false } })
  expect(mockSetLimitMaxSpeed.mock.calls[0][0]).toEqual(false)
})

// -----------------------------------------------------------------------------
test('When the selection type is changed', () => {
  const mockSetSelectionTypeById = jest.fn()
  const component = shallow(
    <CoverageInitializer isSuperUser
      activeSelectionModeId={SelectionModes.SELECTED_AREAS}
      selectionModes={selectionModes}
      coverageType={'location'}
      groupKey={'serviceArea'}
      useMarketableTechnologies
      useMaxSpeed
      coverageReport={{}}
      setSelectionTypeById={mockSetSelectionTypeById}
    />
  )
  component.find('#selectCoverageInitializerSelectionType').simulate('change', { target: { value: SelectionModes.SELECTED_ANALYSIS_AREAS } })
  expect(mockSetSelectionTypeById.mock.calls[0][0]).toEqual(SelectionModes.SELECTED_ANALYSIS_AREAS)
})
