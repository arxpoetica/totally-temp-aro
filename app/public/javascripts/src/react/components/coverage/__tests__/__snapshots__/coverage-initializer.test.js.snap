// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the logged in user is a SuperUser 1`] = `
<div>
  <table
    className="table table-sm table-striped sidebar-options-table"
    id="table-coverage-initializer"
    ng-if="$ctrl.isSuperUser"
  >
    <tbody>
      <tr>
        <td>
          Coverage Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageType"
            onChange={[Function]}
          >
            <option
              value="census_block"
            >
              Form 477
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Save coverage in
        </td>
        <td>
          <select
            className="form-control"
            id="selectGroupKeyType"
            onChange={[Function]}
          >
            <option
              value="serviceArea"
            >
              Service Areas
            </option>
            <option
              value="networkNode"
            >
              Network Nodes
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Limit To Marketable Technology
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMarketableTechnologies"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Limit To Max Equipment Speed
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMaxSpeed"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Site Assignment
        </td>
        <td>
          <select
            className="form-control"
            onChange={[Function]}
            value="Proximity"
          >
            <option
              value="Proximity"
            >
              Proximity
            </option>
            <option
              value="Incremental"
            >
              Incremental
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selection Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageInitializerSelectionType"
            onChange={[Function]}
          >
            <option
              value="SELECTED_AREAS"
            >
              Service Areas
            </option>
            <option
              value="SELECTED_LOCATIONS"
            >
              Locations
            </option>
            <option
              value="SELECTED_ANALYSIS_AREAS"
            >
              Analysis Areas
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selected Geographies
        </td>
        <td>
          <div>
            (no items selected)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`When the logged in user is not a SuperUser 1`] = `
<div>
  <div
    className="alert alert-danger"
    style={
      Object {
        "marginTop": "60px",
      }
    }
  >
    You must be a system-wide superuser in order to use coverage reports.
  </div>
</div>
`;

exports[`When we do not have an existing report 1`] = `
<div>
  <table
    className="table table-sm table-striped sidebar-options-table"
    id="table-coverage-initializer"
    ng-if="$ctrl.isSuperUser"
  >
    <tbody>
      <tr>
        <td>
          Coverage Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageType"
            onChange={[Function]}
          >
            <option
              value="census_block"
            >
              Form 477
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Save coverage in
        </td>
        <td>
          <select
            className="form-control"
            id="selectGroupKeyType"
            onChange={[Function]}
          >
            <option
              value="serviceArea"
            >
              Service Areas
            </option>
            <option
              value="networkNode"
            >
              Network Nodes
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Limit To Marketable Technology
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMarketableTechnologies"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Limit To Max Equipment Speed
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMaxSpeed"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Site Assignment
        </td>
        <td>
          <select
            className="form-control"
            onChange={[Function]}
            value="Proximity"
          >
            <option
              value="Proximity"
            >
              Proximity
            </option>
            <option
              value="Incremental"
            >
              Incremental
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selection Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageInitializerSelectionType"
            onChange={[Function]}
          >
            <option
              value="SELECTED_AREAS"
            >
              Service Areas
            </option>
            <option
              value="SELECTED_LOCATIONS"
            >
              Locations
            </option>
            <option
              value="SELECTED_ANALYSIS_AREAS"
            >
              Analysis Areas
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selected Geographies
        </td>
        <td>
          <div>
            (no items selected)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`When we have an existing report 1`] = `
<div>
  <table
    className="table table-sm table-striped sidebar-options-table"
    id="table-coverage-initializer"
    ng-if="$ctrl.isSuperUser"
  >
    <tbody>
      <tr>
        <td>
          Coverage Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageType"
            onChange={[Function]}
          >
            <option
              value="census_block"
            >
              Form 477
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Save coverage in
        </td>
        <td>
          <select
            className="form-control"
            id="selectGroupKeyType"
            onChange={[Function]}
          >
            <option
              value="serviceArea"
            >
              Service Areas
            </option>
            <option
              value="networkNode"
            >
              Network Nodes
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Limit To Marketable Technology
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMarketableTechnologies"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Limit To Max Equipment Speed
        </td>
        <td>
          <input
            className="checkboxfill"
            id="chkLimitMaxSpeed"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Site Assignment
        </td>
        <td>
          <select
            className="form-control"
            onChange={[Function]}
            value="Proximity"
          >
            <option
              value="Proximity"
            >
              Proximity
            </option>
            <option
              value="Incremental"
            >
              Incremental
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selection Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageInitializerSelectionType"
            onChange={[Function]}
          >
            <option
              value="SELECTED_AREAS"
            >
              Service Areas
            </option>
            <option
              value="SELECTED_LOCATIONS"
            >
              Locations
            </option>
            <option
              value="SELECTED_ANALYSIS_AREAS"
            >
              Analysis Areas
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selected Geographies
        </td>
        <td>
          <div>
            (no items selected)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="disable-sibling-controls"
    ng-if="$ctrl.isSuperUser && $ctrl.coverageReport"
  />
</div>
`;

exports[`When we have supplied all parameters 1`] = `
<div>
  <table
    className="table table-sm table-striped sidebar-options-table"
    id="table-coverage-initializer"
    ng-if="$ctrl.isSuperUser"
  >
    <tbody>
      <tr>
        <td>
          Coverage Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageType"
            onChange={[Function]}
            value="location"
          >
            <option
              value="census_block"
            >
              Form 477
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Save coverage in
        </td>
        <td>
          <select
            className="form-control"
            id="selectGroupKeyType"
            onChange={[Function]}
          >
            <option
              value="serviceArea"
            >
              Service Areas
            </option>
            <option
              value="networkNode"
            >
              Network Nodes
            </option>
            <option
              value="location"
            >
              Locations
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Limit To Marketable Technology
        </td>
        <td>
          <input
            checked={true}
            className="checkboxfill"
            id="chkLimitMarketableTechnologies"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Limit To Max Equipment Speed
        </td>
        <td>
          <input
            checked={true}
            className="checkboxfill"
            id="chkLimitMaxSpeed"
            onChange={[Function]}
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Site Assignment
        </td>
        <td>
          <select
            className="form-control"
            onChange={[Function]}
            value="Proximity"
          >
            <option
              value="Proximity"
            >
              Proximity
            </option>
            <option
              value="Incremental"
            >
              Incremental
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selection Type
        </td>
        <td>
          <select
            className="form-control"
            id="selectCoverageInitializerSelectionType"
            onChange={[Function]}
            value="SELECTED_AREAS"
          >
            <option
              value="SELECTED_AREAS"
            >
              Service Areas
            </option>
            <option
              value="SELECTED_LOCATIONS"
            >
              Locations
            </option>
            <option
              value="SELECTED_ANALYSIS_AREAS"
            >
              Analysis Areas
            </option>
          </select>
        </td>
      </tr>
      <tr>
        <td>
          Selected Geographies
        </td>
        <td>
          <div>
            (no items selected)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="disable-sibling-controls"
    ng-if="$ctrl.isSuperUser && $ctrl.coverageReport"
  />
</div>
`;
